cpp/models/abm/world.cpp

```diff
@@ -65,32 +78,32 @@ void World::evolve(TimePoint t, TimeSpan dt)
 
 void World::interaction(TimePoint t, TimeSpan dt)
 {
+    const uint32_t num_persons = static_cast<uint32_t>(m_persons.size());
     PRAGMA_OMP(parallel for)
-    for (auto i = size_t(0); i < m_persons.size(); ++i) {
-        auto&& person     = m_persons[i];
-        auto personal_rng = Person::RandomNumberGenerator(m_rng, *person);
-        person->interact(personal_rng, t, dt, parameters);
+    for (uint32_t i = 0; i < num_persons; ++i) {
+        interact(i, t, dt);
```

Comment:
In this case I would usually keep the `i`, since it is used in a very limited scope and doesn't do anything special here. But the loop in migration() should probably have a clearer iterator as it is used in a deeply nested context. I'll change both.